rename_package.sh

```diff
@@ -0,0 +1,141 @@
+#!/bin/bash
+
+HELP_MESSAGE="Usage: $0 OLD_PACKAGE_NAME NEW_PACKAGE_NAME [--dry-run]
+
+Rename a package by replacing all text occurences of the old package name with
+the new package name and renaming all matching files and directories.
+
+The first positional argument is the old package name to replace.
+
+The second positional argument is the desired new package name.
+
+Use --dry-run to prevent any filesystem changes while testing the usage.
+
+This script replaces all text occurences of OLD_PACKAGE_NAME with NEW_PACKAGE_NAME
+in all files in the following search paths:
+  - ./.devcontainer.json
+  - ./aica-package.toml
+  - ./source/**
```

Comment:
Personally not convinced by this script and the potential associated dangers (e.g. what if a user needs to keep the name intact in source for compatibility).

Reply:
you are right about that of course, but it's clearly a burden on the user to know all the places to rename the template package when they make it into their own component. I would be happy to know if you can think of better alternatives. 

If you prefer, I can remove the public option to configure the source name (i.e. make it only work the first time for `template_component_package` to `X`) `./rename-package.sh bar` instead of `./rename-package.sh foo bar`.

Reply:
> I would be happy to know if you can think of better alternatives.

The only safe way I can think would be to do something like `npm init` does, where you configure the name and such when creating the project instead of later.

> `./rename-package.sh bar` instead of `./rename-package.sh foo bar`

I think that's better.